tests/wormhole/test_silicon_driver_wh.cpp

```diff
@@ -644,3 +655,85 @@ TEST(SiliconDriverWH, VirtualCoordinateBroadcast) {
     }
     device.close_device();    
 }
+
+
+/**
+ * This is a basic DMA test -- not using the PCIe controller's DMA engine, but
+ * rather using the ability of the NOC to access the host system bus via traffic
+ * to the PCIe block.
+ *
+ * sysmem means memory in the host that has been mapped for device access.  It
+ * is currently one or more 1G huge pages, although this may change.
+ *
+ * 1. Fills sysmem with a random pattern.
+ * 2. Uses PCIe block on WH to read sysmem into buffer.
+ * 3. Verifies that buffer matches sysmem.
+ * 4. Fills buffer with a random pattern.
+ * 5. Uses PCIe block on WH to write buffer into sysmem.
+ * 6. Verifies that sysmem matches buffer.
+ *
+ * This uses a small size for speed purposes.
+ *
+ * If/when we move to using IOMMU to map userspace memory for device access,
+ * the technique below is a straightforward way to test that hardware can access
+ * the buffer(s).
+ */
+TEST(SiliconDriverWH, SysmemTestWithPcie) {
+    auto target_devices = get_target_devices();
+
+    // Why is this required by the tt_SiliconDevice constructor?!
+    std::unordered_map<std::string, std::int32_t> dynamic_tlb_config = {
+        { "REG_TLB", tt::umd::wormhole::REG_TLB }
+    };
+
+    tt_SiliconDevice device(test_utils::GetAbsPath("tests/soc_descs/wormhole_b0_8x10.yaml"),
+                            test_utils::GetClusterDescYAML(),
+                            target_devices,
+                            1,  // one "host memory channel", currently a 1G huge page
+                            dynamic_tlb_config,
+                            false, // skip driver allocs - no (don't skip)
+                            true,  // clean system resources - yes
+                            true); // perform harvesting - yes
+
+    // PCIe core is at (x=0, y=3) on Wormhole NOC0.
+    const size_t PCIE_X = 0;    // NOC0
+    const size_t PCIE_Y = 3;    // NOC0
+    const size_t test_size_bytes = 0x4000;
```

Comment:
Is there some reason why we are doing this with `0x4000` bytes? Whether it is random or specific for the test, it would be nice to have a comment here

Reply:
The technique in the test is slow.  0x4000 is somewhat arbitrary - enough that it demonstrates that the mechanism is working, but small enough that the test finishes in less than one second.  I added a comment.